### ISSUE TITLE
SoftMax Regression 发生溢出

### ISSUE THREAD

**Issue by idlebo** [Hzfinfdu/FDU-PRML-2023Fall#4]
我的 softmax 在计算 loss 和梯度的时候都发生了数据溢出的情况，请问有什么解决方法吗（我尝试调小 learning rate，调小 reg，还有调小 num_iters 但是这些都会使 softmax 的效果下降，我尝试改用更高精度的数据类型也依然发生溢出😢😭）

**Comment by Hzfinfdu** [comment]
有同学遇到过类似的问题吗？是会爆nan吗？还是inf？

可以尝试排查初始化的影响，并在优化的过程中每一步打印weight，查看优化轨迹以寻找可能的原因

**Comment by idlebo** [comment]
nan 和 inf 都会有

好的，我后面调试一下

**Comment by Hzfinfdu** [comment]
其他可能的问题还包括loss设计有问题，因此监督信号可能会很混乱导致的

**Comment by idlebo** [comment]
我调了一下，发现好像就是我的 learning rate 和 reg 过大导致的。

至于为什么精度很差，把  learning rate 和 reg 调大发生溢出反而会得到更好的效果我还不是很清楚。
